### PR TITLE
New doc

### PR DESCRIPTION
**Pull request by fhamon**
I added jsdoc as a dev dependency and commented all the functions for the framework. Feel free to ask for changes. ✌️

### CODE REVIEW

File: Gruntfile.js [hunk 1]
```diff
@@ -258,6 +267,7 @@ module.exports = function fxGruntConfig (grunt) {
 		});
 		
 		// Default task.
+		grunt.registerTask('doc', ['jsdoc']);
 		grunt.registerTask('test', ['connect', 'qunit']);
 		grunt.registerTask('dev', ['jshint', 'jscs', 'complexity']);
 		grunt.registerTask('build', [
```
Comment (nitriques) [inline Gruntfile.js]: build should call doc :+1:

File: src/app-callback.js [hunk 2]
```diff
@@ -1,14 +1,25 @@
 /**
- * @author Deux Huit Huit
- *
- * App Callback functionnality
- *
- */
+* @author Deux Huit Huit
```
Comment (nitriques) [inline src/app-callback.js]: \* are not properly aligned.

File: src/app.js [hunk 3]
```diff
@@ -148,11 +200,17 @@
 			return $();
 		};
 		
+		/**
+		 * 
+		 */
 		var enterLeave = function () {
```
Comment (nitriques) [inline src/app.js]: Missing doc

File: src/app.js [hunk 4]
```diff
@@ -123,15 +167,23 @@
...
+	 * @this App
+	 */
 	var gotoPage = function (obj, previousPoppedUrl) {
 		var nextPage;
 		var route = '';
 		
+		/**
+		 * Try to parse the data in jQuery to be sure it's valid
+		 * @param {String} data response data
+		 * @returns {jQueryElement}
```
Comment (nitriques) [inline src/app.js]: Should be only `jQuery`

File: src/app.js [hunk 5]
```diff
@@ -123,15 +167,23 @@
...
 	/**
-	* Change the current page to the requested route
-	* Do nothing if the current page is already the requested route
-	*/
+	 * Change the current page to the requested route
+	 * Do nothing if the current page is already the requested route
+	 * @param {String} obj page requested
+	 * @param {String} previousPoppedUrl
+	 * @this App
+	 */
```
Comment (nitriques) [inline src/app.js]: We need to find a way to add in the doc the fact that this method will publish App.modules.notify('page.leave', {page: previousPage});

I've never been able to do it, but this is THE MOST important information to have

File: src/globals.js [hunk 6]
```diff
@@ -1,17 +1,15 @@
 /**
- * @author Deux Huit Huit
- */
-
-/**
- * General customization
+ * General customization alongside the framework
+ *
+ * @author Deux Huit Huit <http://deuxhuithuit.com>
+ * @license MIT <http://deuxhuithuit.mit-license.org>
```
Comment (nitriques) [inline src/globals.js]: License links must be https

File: Gruntfile.js (hunk 1 above)
Comment (fhamon) [reply Gruntfile.js]: 👍

File: src/app.js (hunk 4 above)
Comment (fhamon) [reply src/app.js]: 👍

**Comment by fhamon** [comment]
@nitriques You can review at this time 😁

**Pull request merged**